### DISCUSSION TITLE
How can I line-wrap code blocks rather than having to scroll it.

### DISCUSSION THREAD

**Discussion by NunoLemos101** [gonzalezreal/swift-markdown-ui#260] [Q&A] [answered]
So basically I have built an application natively for Android and here is a screenshot of how the Markdown is rendered in Android

Now I'm trying to render that same markdown (it comes from a backend) in SwiftUI and the code block doesn't line-wrap. Instead it forces the user to scroll horizontally.

I would like to know if It's possible to achieve this result in SwiftUI. Make the code blocks line-wrap.

Thank you,

![image](https://github.com/gonzalezreal/swift-markdown-ui/assets/58612710/b206269a-ac4f-438d-b61c-7ebbde28b8a5)

**Comment by gonzalezreal** [comment, marked as answer]
Hi @NunoLemos101,

You can use the [`markdownBlockStyle(_:body:)`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.2.0/documentation/markdownui/tablecellconfiguration/label-swift.struct/markdownblockstyle(_:body:)-2sx0c) modifier to replace a specific block style on the current [`Theme`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.2.0/documentation/markdownui/theme).

To achieve the desired effect, you should replace the current [`codeBlock`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.2.0/documentation/markdownui/theme/codeblock) style with a new one that does not use a `ScrollView`. Here is an example:

```swift
Markdown("...")
  .markdownBlockStyle(\.codeBlock) { configuration in
    configuration.label
      .fixedSize(horizontal: false, vertical: true)
      .relativeLineSpacing(.em(0.225))
      .markdownTextStyle {
        FontFamilyVariant(.monospaced)
        FontSize(.em(0.85))
      }
      .padding(16)
      .background(Color(.secondarySystemBackground))
      .clipShape(RoundedRectangle(cornerRadius: 6))
      .markdownMargin(top: 0, bottom: 16)
  }
```

<img src="https://github.com/gonzalezreal/swift-markdown-ui/assets/373190/181f4301-978d-496c-a605-a88e198551ff" width="375"/>

**Reply by NunoLemos101** [reply]
Hi @gonzalezreal 

I managed to do it using `ScrollView(.vertical)` but your approach is certainly better.

Thank you
Nuno Lemos
